core/src/apps/common/sd_salt.py

```diff
@@ -57,171 +42,37 @@ class SdProtectCancelled(Exception):
 
 
 async def request_sd_salt(
-    ctx: Optional[wire.Context], salt_auth_key: bytes
-) -> bytearray:
-    salt_path = _get_salt_path()
-    new_salt_path = _get_salt_path(True)
-
+    ctx: wire.GenericContext = wire.DUMMY_CONTEXT
+) -> Optional[bytearray]:
     while True:
-        sd = io.SDCard()
-        fs = io.FatFS()
-        while not sd.power(True):
-            await _insert_card_dialog(ctx)
-
-        try:
-            fs.mount()
-            salt = _load_salt(fs, salt_auth_key, salt_path)
-            if salt is not None:
-                return salt
-
-            # Check if there is a new salt.
-            salt = _load_salt(fs, salt_auth_key, new_salt_path)
-            if salt is not None:
-                # SD salt regeneration was interrupted earlier. Bring into consistent state.
-                # TODO Possibly overwrite salt file with random data.
-                try:
-                    fs.unlink(salt_path)
-                except OSError:
-                    pass
-
-                try:
-                    fs.rename(new_salt_path, salt_path)
-                except OSError:
-                    error_dialog = _write_failed_dialog(ctx)
-                else:
-                    return salt
-            else:
-                # No valid salt file on this SD card.
-                error_dialog = _wrong_card_dialog(ctx)
-        finally:
-            fs.unmount()
-            sd.power(False)
-
-        await error_dialog
-
-
-async def set_sd_salt(
-    ctx: Optional[wire.Context], salt: bytes, salt_tag: bytes, new: bool = False
-) -> None:
-    salt_path = _get_salt_path(new)
-
-    while True:
-        sd = io.SDCard()
-        while not sd.power(True):
-            await _insert_card_dialog(ctx)
-
-        try:
-            fs = io.FatFS()
-            fs.mount()
-            fs.mkdir("/trezor", True)
-            fs.mkdir(_get_device_dir(), True)
-            with fs.open(salt_path, "w") as f:
-                f.write(salt)
-                f.write(salt_tag)
-            break
-        except Exception:
-            fs.unmount()
-            sd.power(False)
-            await _write_failed_dialog(ctx)
-
-    fs.unmount()
-    sd.power(False)
-
-
-async def stage_sd_salt(
-    ctx: Optional[wire.Context], salt: bytes, salt_tag: bytes
-) -> None:
-    await set_sd_salt(ctx, salt, salt_tag, True)
-
-
-async def commit_sd_salt(ctx: Optional[wire.Context]) -> None:
-    salt_path = _get_salt_path()
-    new_salt_path = _get_salt_path(True)
-
-    sd = io.SDCard()
-    fs = io.FatFS()
-    if not sd.power(True):
-        raise OSError
-
-    try:
-        fs.mount()
-        # TODO Possibly overwrite salt file with random data.
+        ensure_sd_card(ctx)
         try:
-            fs.unlink(salt_path)
+            return storage.sd_salt.load_sd_salt()
+        except storage.sd_salt.WrongSdCard:
+            if not await _wrong_card_dialog(ctx):
+                raise SdProtectCancelled
```

Comment:
So for example here, shouldn't we raise wire.ProcessError if ctx is not DUMMY_CONTEXT? Or maybe we should reraise WrongSdCard and translate later?